### ISSUE TITLE
hack/update-codegen.sh is not useful

### ISSUE THREAD

**Issue by zffocussss** [kubernetes-sigs/scheduler-plugins#445]
Hi team:
  as I want to add a new out-of-tree plugin,after preparing the data structure, issuing hack/update-codegen.sh,no deepcopy is generated.can you help me?

**Comment by JingbenShi668** [comment]
> Hi team: as I want to add a new out-of-tree plugin,after preparing the data structure, issuing hack/update-codegen.sh,no deepcopy is generated.can you help me?

You need to notice the storage location of your cloned repo. For example, if you attempt to generate code for your out-of-tree plugin in Linux environment, the cloned repo location should be like $GOPATH/src/sigs.k8s.io/scheduler-plugins. 

In scheduler-plugins/hack/update-codegen.sh file, the prefix of the directory positions in bash command in the red location may be custom. 
<img width="748" alt="image" src="https://user-images.githubusercontent.com/39040502/200188227-2faeedf7-c6a4-42ab-ab19-cd33f1b0fded.png">

I just use the default repo name and it is scheduler-plugins. I can successfully  generate code for my custom plugin. But it seems that some generated code is wrong in the scheduler-plugins/pkg/apis/config/v1beta1/zz_generated.defaults.go file. 
In other words, it doesn't generate setDefaultValue function.

**Comment by Huang-Wei** [comment]
@zffocussss did you place your repo under GOPATH? and as @JingbenShi668 mentioned, you may need to tweak some literals in that script if you're running at a different folder.

**Comment by zffocussss** [comment]
> 

thanks,I will try as your advice

**Comment by zffocussss** [comment]
> 

@Huang-Wei no,I just put my repo in other place,without relation to go env.

**Comment by zwpaper** [comment]
after https://github.com/kubernetes-sigs/scheduler-plugins/pull/481 get merged, scheduler plugins can now work outside of GOPATH.

this issue should be ok to close, @Huang-Wei @zffocussss

**Comment by Huang-Wei** [comment]
/close

Workaround for this issue: leverage #481, or place the repo under GOPATH.
